docs/guide/database-queries.md

```diff
@@ -0,0 +1,273 @@
+- `.relationships(false)`: This builder accepts `false` or an array of relationship field names to avoid or limit relationship data loading onto the request response (saving some processing when it's not needed).
+
+As we can see here, query builders apply a wide variety of effects. See the [query builder reference page](/reference/query-builders.md) for a full list of query builders and how to use them.
+
+Let's take another look at the builders in the example above.
+
+```javascript
+self.find(req, criteria)
+  .sort({ createdAt: -1 })
+  .limit(5)
+```
+
+This example is using two query builders, `sort` and `limit`. As described above, `sort({ createdAt: -1})` sorts the queried documents in descending order on their `createdAt` properties (most recent first). `limit(5)` then ensures the request will return no more than five documents.
+
+::: note
+Many aspects of querying the database are the same for both page and piece queries. One thing that is different are certain query builders that only apply to requests by the page module. These builders do things like update how results should include "child" or "ancestor" pages from the page hierarchy.
+
+There is [a section for these builders](/reference/query-builders.md#page-document-query-builders) in the reference page. There is a similar section with a builder that only works on *image* piece queries as well.
+:::
+
+### Projections: Reducing returned document data
+
+One very useful query builder is `.project()`. We often know what specific data properties we want from a query and it is rarely *every single property*. By applying a query projection we reduce density of the data returned, making the response lighter, faster, and easier work with during development.
+
+```
+query
+  .project({
+    title: 1,
+    _author: 1,
+    publishedAt: 1
+  })
+```
+
+The `project` builder is one that uses a MongoDB syntax. The object passed as an argument should include schema field names for the document type we're querying with each set to `1`. This tells the database that we only want these fields in each document we get back.
+
+The `_id` property is always included no matter what projection is used.
+
+::: note
+In Apostrophe 2 this builder was named `projection`.
+:::
+
+### Paginating query results
+
+There are a few query builders that can work together to "paginate" query results. Sometimes that is better than receiving an unknown large number of results all at once.
+
+```javascript
+query
+  .perPage(20)
+  .page(2)
+```
+
+This example tells the query that its results should be returned in a group of no more than 20 (`.perPage(20)`) and that we want the second group of results based on the active sort order (`.page(2)`). These builders are used in the Apostrophe REST APIs and can be very helpful when dealing with large amounts of content. They use two other builders internally (`limit` and `skip`), but can be easier to use for common situations.
+
+### Adding your own query builders
```

Comment:
Before you say this, you should say that a query builder is automatically added for most schema field types (if it has an `addQueryBuilder` method in its field type definition, or extends one that does, then you can automatically use it as a query builder). This is a big part of what we mean when we say we save people effort over mastering MongoDB.

For relationship fields, there are several query builders added. If the field name is `_products`, then:

* `_products(product._id)` matches only documents related to the product with the given `_id`. You can also pass an array of product ids, in which case only documents related to *at least one of* those products will match.
* `products(product.slug)` (note no `_`) matches documents related to the product with that slug. You can also pass an array of product slugs, in which case only documents related to *at least one of* those products will match.
* `_productsAnd([ id, id, id...])` matches only documents related to *all* of the specified product ids (just one is not enough).
* `productsAnd([ slug, slug, slug...])` (note no `_`) matches only documents related to *all* of the specified product slugs (just one is not enough).

The main advantage of the query builders based on `_id` is that the `_id` can never change. The main advantage of the query builders based on slugs is that they are more user-friendly if they appear in the query string fo the URL.